lib/rouge/lexers/csvs.rb

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class CSVS < RegexLexer
+      tag 'csvs'
+      title "csvs"
+      desc 'The CSV Schema Language (http://digital-preservation.github.io/csv-schema/)'
+      mimetypes 'text/x-csvssrc'
+      filenames '*.csvs'
+
+      def self.detect?(text)
+        return true if text.shebang? 'csvs'
+      end
+
+      def self.constants
+        @constants ||= Set.new %w(nil false true)
+      end
+
+      def self.builtins
+        @builtins ||= Set.new %w(
+          args call clone do doFile doString else elseif for if list
+          method return super then
+        )
+      end
+
+      state :root do
+        rule /\s+/m, Text
+        rule %r(//.*?\n), Comment::Single
+        rule %r(#.*?\n), Comment::Single
+        rule %r(/(\\\n)?[*].*?[*](\\\n)?/)m, Comment::Multiline
+        rule %r(/[+]), Comment::Multiline, :nested_comment
+
+        rule /"(\\\\|\\"|[^"])*"/, Str
+
+        rule %r(:?:=), Keyword
+        rule /[()]/, Punctuation
+
+        rule %r([-=;,*+><!/|^.%&\[\]{}]), Operator
+
+        rule /[A-Z]\w*/, Name::Class
+
+        rule /[a-z_]\w*/ do |m|
+          name = m[0]
```

Comment:
You may as well leave this off. It's fine to just pass `m[0]` to the methods below.